From 1ec95b448a433d7ad428958dec6756b1dbdef072 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 25 Jul 2019 13:09:30 +0700
Subject: [PATCH 173/326] arm64: dts: renesas: r8a774b1: Add INTC-EX device
 node

Adds a device node for the Interrupt Controller for External
Devices (INTC-EX) on RZ/G2N.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774b1.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774b1.dtsi b/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
index 3741fa5..3ac3317 100644
--- a/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774b1.dtsi
@@ -204,6 +204,22 @@
 			reg = <0 0xe6060000 0 0x50c>;
 		};
 
+		intc_ex: interrupt-controller@e61c0000 {
+			compatible = "renesas,intc-ex-r8a774b1", "renesas,irqc";
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0 0xe61c0000 0 0x200>;
+			interrupts =	<GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH
+					 GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 407>;
+			power-domains = <&sysc R8A774B1_PD_ALWAYS_ON>;
+			resets = <&cpg 407>;
+		};
+
 		tsc: thermal@e6198000 {
 			reg =	<0 0xe6198000 0 0x100>,
 				<0 0xe61a0000 0 0x100>,
-- 
1.9.1

